app/uk/gov/hmrc/gform/validation/ComponentsValidator.scala

```diff
@@ -116,7 +116,15 @@ class ComponentsValidator[D <: DataOrigin, F[_]: Monad](
   private def produceValidationError(
     message: SmartString
   ): ValidatedType[Unit] =
-    Map(formComponent.modelComponentId -> Set(message.value)).invalid
+    formComponent.`type` match {
```

Comment:
Can we not do it for any multivalue component? I have just implemented calendarDate and it also needs a similar change

Reply:
Done. Please check